### PR TITLE
Fix array slice crash when used as a bare value (#5132)

### LINKED ISSUE #5132: Compiling a sliced unpacked array leads to `Unexpected Call` in `V3AstNodeExpr`
```sv
module m;
  byte mem[8] = '{1, 2, 3, 4, 5, 6, 7, 8};
  initial begin
    $display("%p", mem[0:3]);
    $finish;
  end
endmodule
```

```
verilator --binary m.sv
```
```
%Error: m.sv:5:23: Internal: Unexpected Call
    5 |     $display("%p", mem[0:3]);
      |                       ^
%Error: Internal Error: m.sv:5:23: ../V3AstNodeExpr.h:4658: Unexpected Call
    5 |     $display("%p", mem[0:3]);
      |                       ^
```
Expected output:
```
'{1, 2, 3, 4}
```

v5.024-56-g5f85c8942 5f85c8942 on ArchLinux

### PR DESCRIPTION
**Pull request by MrPoloGit**
Fixes https://github.com/verilator/verilator/issues/5132.

An array slice (arr[a:b]) used directly as a value, e.g. a $display argument, hit an internal error, instead of the assignment/comparison contexts Verilator already supported for slices. This has now been handled.

### CODE REVIEW

File: test_regress/t/t_slice_display.v [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+`define stop $stop
+`define checkh(gotv,expv) do if ((gotv) !== (expv)) begin $write("%%Error: %s:%0d:  got='h%x exp='h%x\n", `__FILE__,`__LINE__, (gotv), (expv)); `stop; end while(0)
+// verilog_format: on
+
+// Issue #5132: an array slice used as a bare value (not the LHS/RHS of an
+// assignment) hit an internal error.
+module t;
+  byte mem[8] = '{1, 2, 3, 4, 5, 6, 7, 8};
+  byte dmem[7:0] = '{10, 20, 30, 40, 50, 60, 70, 80};
+  byte sub[4];
```
Comment (wsnyder) [inline test_regress/t/t_slice_display.v:17]: Please add checks for non-zero low indexes, and opposite endian e.g. `[3:6]`

Please also have something wide e.g. `logic [72:0] memwide[6:3]`

**Review by wsnyder — COMMENTED** [review]
Thanks for the PR. When you think everything is changed, and passing please re-request review.

**Comment by MrPoloGit** [comment]
@wsnyder there was an issue when installing for one of the GA tests, `Error: Process completed with exit code 4.` . It doesn't seem like I have permissions to rerun GA tests, I could push again but it would cause everything to run again. Do you think you could maybe rerun the failed one?

**Comment by MrPoloGit** [comment]
Rebuilding locally, I will add 2 more tests after its done.

**Comment by MrPoloGit** [comment]
Ready now.

**Review by HUNT-001 — COMMENTED** [review]
Nice fix! Crashing on bare slice expressions is definitely a tricky edge case, so it's awesome to see #5132 being addressed.

I took a look through the changes and had a few quick thoughts / edge cases that might be worth double-checking:

Side effects in index: If the slice index expression includes a function call or something with side effects (e.g., arr[get_idx():0]), does the AST cleanup retain that call or could it get accidentally dropped?

Unpacked or Dynamic arrays- Does this logic hold up cleanly for unpacked arrays and dynamic arrays without hitting issues later in passes like V3Width or V3Const?

Nested slices: Worth checking if multi-dimensional or nested slices (like arr[1:0][3:0];) behave as expected with this fix.

Regression Test:
Do you have a small test case added under test_regress/t/ for this? Just something simple like: 
```module t;
  logic [7:0] arr [3:0];
  initial begin
    arr[2:1]; // bare slc eval
    $display("Pass");
    $finish;
  end
endmodule```
Having that in ```test_regress/``` will make sure we don't accidentally regress on this down the road.
